File: Lab7/ExpressionCalcDLL/ManualSQRT.cs
namespace ExpressionCalcDLL
{
    public class ManualSQRT
    {
        private double number;

        public double GetNumber()
            { return number; }

        public ManualSQRT(double value)
        {
            if (value < 0)
            {
                throw new ArgumentException($"Inacceptable square value: {value}");
            }

            if (value == 0)
            {
                number = 0;
            }
            else
            {
                number = Math.Pow(value, 0.5);
            }
        }
    }
}

File: Lab7/ExpressionCalcStatic/ManualSQRT.cpp
#include "ManualSQRT.h"
#include "math.h"
#include <format>
#include <iostream>
#include <string>

using namespace std;

ManualSQRT::ManualSQRT(double value)
{
    if (value < 0)
    {

        string error = "Inacceptable square value:" + to_string(value);
        throw runtime_error(error);
    }

    if (value == 0)
    {
        number = 0;
    }
    else
    {
        number = pow(value, 0.5);
    }
}

double ManualSQRT :: GetNumber()
{
    return number;
}